Deprecation of normalizePath to fix it in the new method normalizeFullPath

AlPri78 opened the pull request:
#12131  

Everything in the title and in the issue.

It seems that other languages change only slashes, it's what I did. It could be possible to suppress double slashes, but because it's not very problematic and rare , so I take the option to not manage that. Tell me what do you think, perhaps other kind of work to do in this new method.

I also added some directives for a correct code analysis.

It misses documentation. I don't know if deprecation is automatically calculated or not. Please tell me and I will do the necessary.

AlPri78

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.
Added some directives for code analysis.

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.
Added some directives for code analysis (Change directive writing for Travis CI).

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.
Added some directives for code analysis (Change directive writing x2 for Travis CI).

markstory commented on src/Filesystem/Folder.php:
```diff
@@ -354,12 +354,28 @@ public static function isRegisteredStreamWrapper($path)
      * @param string $path Path to check
      * @return string Set of slashes ("\\" or "/")
+     *
+     * @deprecated 3.6.5 This method will be removed in 4.0.0. Use correctSlashFor() instead.
```
Deprecations should emit runtime warnings with `deprecationWarning()` in addition to the documentation tag.

markstory commented on tests/TestCase/Filesystem/FolderTest.php:
```diff
@@ -680,23 +680,50 @@ public function testSlashTerm()
         $this->assertEquals('/', $result);
 
         $path = 'C:\\path\\to\\file';
+        /** @noinspection PhpDeprecationInspection */
```
You can use the `$this->deprecated()` helper to swallow deprecation warnings during this test. We generally don't use these comment directives either.

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.
Added some directives for code analysis (Change directive writing x2 for Travis CI).
Added deprecationWarning, and suppress noinspection directive for
deprecation

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.
Added some directives for code analysis (Change directive writing x2 for Travis CI).
Added deprecationWarning, and suppress noinspection directive for
deprecation.
Added deprecated function in Test function.

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.

ADmad commented on src/Filesystem/Folder.php:
```diff
@@ -355,7 +355,7 @@ public static function isRegisteredStreamWrapper($path)
      * @return string Set of slashes ("\\" or "/")
      *
-     * @deprecated 3.6.5 This method will be removed in 4.0.0. Use correctSlashFor() instead.
+     * @deprecated 3.7 This method will be removed in 4.0.0. Use correctSlashFor() instead.
```
Since this patch is targeted to `master` I will go into next `3.6` release hence the version this method will be deprecated in will be `3.6.5`.

AlPri78 replied:
I changed that because of Markstory put the 3.7 milestone yesterday.
Do you ask to change that or is it not correlated ?

alpri49 committed:
Deprecated normalizePath to fix it in the new method normalizeFullPath.

Commit:
Deprecated normalizePath to fix it in the new method normalizeFullPath.

AlPri78 commented:
Hello, I'd need some help to finalize.
Since markstory changes the branch I encounter different errors in Travis but not on my coding. What did I miss ?

markstory commented:
Thank you 👍

markstory commented:
> If possible, may I have just a very little explanation on why I encountered errors on Travis, and you merge my modifications with that... It's just for my potential future contributions!

Travis often has partial failures due to infrastructure issues or timeouts. There isn't much you need to do in those scenarios as we can restart builds and compare individual stages of the matrix with their sibling builds.
